nvflare/fuel/hci/client/fl_admin_api.py

```diff
@@ -199,6 +199,15 @@ def _validate_file_string(self, file: str) -> str:
             )
         return file
 
+    def _validate_sp_string(self, sp_string) -> str:
+        if re.match(
+            r"^((([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9]):[0-9]*:[0-9]*)$",
+            sp_string,
+        ):
+            return sp_string
+        else:
+            raise APISyntaxError("sp_string must be of the format example.com:8002:8003")
+
```

Comment:
Just a note, if any other places is also using this we might want to put this in a common util.